lib/move/index.js

```diff
@@ -35,7 +35,9 @@ function move (source, dest, options, callback) {
   }
 
   function doRename () {
-    if (overwrite) {
+    if (path.resolve(source) === path.resolve(dest)) {
+      setTimeout(callback, 1)
+    } else if (overwrite) {
```

Comment:
> what are your thoughts on setTimeout over setImmediate? It's my understanding setTimeout with an input of 1, still clamps to 15 ms.

I have no strong opinions on it here since the primary goal is to just defer it a bit. Based on that goal [setImmediate](https://nodejs.org/api/timers.html#timers_setimmediate_callback_args) looks like a better call.